pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -140,7 +150,7 @@ public void setSchemaInfoProvider(SchemaInfoProvider schemaInfoProvider) {
     @Override
     public SchemaInfo getSchemaInfo() {
         if (!schemaMap.containsKey(SchemaVersion.Latest)) {
-            return null;
+            return SCHEMA_INFO;
```

Comment:
this is a public method, so may we don't change this method is better

Reply:
So here it looks like this. If AUTO_CONSUME consumer is used to subscribe to a topic, and the topic has no schema, the schemaMap in AutoConsumeSchema will have no key for SchemaVersion.Latest. Then do getSchemaInfo, schemainfo is null. There is no way to upload the AUTO_CONSUNE to the broker. So in the previous commit, if (si == null && schema instanceof AutoConsumeSchema), I create a schemainfo with a schemaType of AUTO_CONSUME. What am I supposed to do here? Do not change getSchemaInfo, still use the previous code?

Reply:
you can add a new method in AutoConsumeSchema `getAutoConsumeSchemaInfo()` use this method instead of getSchemaInfo when the schema type is AUTO_CONSUME

Reply:
1.In AutoConsumeSchema, we don't change getSchemaInfo:
```
    @Override
    public SchemaInfo getSchemaInfo() {
        if (!schemaMap.containsKey(SchemaVersion.Latest)) {
            return null;
        }
        return schemaMap.get(SchemaVersion.Latest).getSchemaInfo();
    }
```
2.Add a new method in AutoConsumeSchema:
```
    public SchemaInfo getAutoConsumeSchemaInfo() {
        return SCHEMA_INFO;
    }
```
3.In ConsumerImpl:
```
    SchemaInfo si = schema.getSchemaInfo();
    if (si != null && (SchemaType.BYTES == si.getType() || SchemaType.NONE == si.getType())) {
        // don't set schema for Schema.BYTES
        si = null;
    } else {
        if (schema instanceof AutoConsumeSchema) {
            si = ((AutoConsumeSchema) schema).getAutoConsumeSchemaInfo();
        }
    }
```
Is that OK?